### PR TITLE
test: logging: sync log handling with log process thread by semaphore

### LINKED ISSUE #27570: up_squared:logging.add.logging.add.async failed
Describe the bug
Running testcase in tests/subsys/logging/log_core_additional/ on up_squared board failed.

To Reproduce

    west build -b up_squared tests/subsys/logging/log_core_additional/
    west flash ~/up_squared.sh
    see error.

Logs and console output

WARNING: no console will be available to OS
error: no suitable video mode found.
*** Booting Zephyr OS build zephyr-v2.3.0-1774-g34859d0d3e40  ***
Running test suite test_log_list

===================================================================
START - test_multiple_backends
Test multiple backends PASS - test_multiple_backends

===================================================================
START - test_log_domain_id
t00:00:00. 0P4A4S,S0 0-0 ]t e[t0_ml<oign_fd>o mlaoign__tieds
=:= =i=n=f=o= =m=e=s=s=a=g=e= =f=o=r= =d=o=m=a=i=n= =i=d= =t=e=s=t=[=0=m=

 =============================
START - test_log_severity
000:
u0 : 0 1 .A0s6s3e,r0t0i0o]n  [a0iml<eidn fa>t  lWoEgS_Tt_eTsOtP:D IiRn/fzoe pmheysrs/atgees[s0/ms
sb[s0y0s:/0l0o:g0g1i.n0g6/3l,o0g0_0c]o r[_1a;d3d3imt<iwornna>l /lsorgc_/tleosgt_:t ewsatr.nci:n2g6 0m:e stseasgte_[o0gm_
oe[v0e0r:i0t0y::0 1(.30 6n3o,t0 0e0q]u a[ 1t;o3 1bma<cekrern>d 1l_ocgb_.tceosutn:t eerr)r
orU nmeexspseacgtee[ 0amm
 unt of messages received by the backend.
 FAIL - test_log_severity

===================================================================
START - test_log_timestamping
Register timestamp function
 PASS - test_log_timestamping

===================================================================
START - test_log_early_logging
Create log message before backend active
Activate backend with context PASS - test_log_early_logging

===================================================================
START - test_log_sync
Logging synchronousely
 SKIP - test_log_sync

===================================================================
START - test_log_thread
Logging buffer is configured to 1024 bytes
Stack size of logging thread is configured byCONFIG_LOG_PROCESS_THREAD_STACK_SIZE: 4096 bytes
 PASS - test_log_thread

===================================================================
Test suite test_log_list failed.

===================================================================
PROJECT EXECUTION FAILED

Environment (please complete the following information):

    OS: Fedora28
    Toolchain: Zephyr-sdk-0.11.3
    Commit ID: 34859d0d3e

### PR DESCRIPTION
**Pull request by mengxianglinx**
sync log handling with log process thread by semaphore.

Fixes #27570

Signed-off-by: Meng xianglin <xianglinx.meng@intel.com>

### CODE REVIEW

**Review by LeiW000 — APPROVED** [review]
Thanks for fixing it.

**Review by enjiamai — APPROVED** [review]

File: tests/subsys/logging/log_core_additional/src/log_test.c [hunk 1]
```diff
@@ -34,6 +35,8 @@ struct backend_cb {
 	uint16_t exp_severity[4];
 	/* inform put() to check domain_id of message */
 	bool check_domain_id;
+	/* in asnyc mode, make sure all logs have been handled */
```
Comment (sjg20) [inline tests/subsys/logging/log_core_additional/src/log_test.c:38]: async

Also this comment should explain what this field is actually for

**Review by sjg20 — DISMISSED** [review]
LGTM with comment update

File: tests/subsys/logging/log_core_additional/src/log_test.c (hunk 1 above)
Comment (mengxianglinx) [reply tests/subsys/logging/log_core_additional/src/log_test.c]: Thank you @sjg20 ! updated.

**Comment by LeiW000** [comment]
@sjg20 , please approve the PR if you have no other comment.

**Review by nashif — DISMISSED** [review]
please fix the title of the commit, put the issue being fixed in the body, not the commit title.

**Comment by LeiW000** [comment]
@nashif , xianglin has made the change as what you suggested. Please take look at it again.

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**